Add in Jest and React Testing ESLint Plugins

bryanpizzillo opened the pull request:
There will be 4 commits in total, one for turning on Jest linting, one for fixing Jest linting errors, one for turning on react-testing linting, one for fixing the errors.

This closes #33

bryanpizzillo committed:
(#33) Added ESLint Rules for Tests

- Updated testing-library packages; react-testing
  lint rules required newer package.
- Added eslint-plugin-testing-library
- Added eslint-plugin-jest-dom
- Added eslint-plugin-jest
- Enabled recommended rule in all, as well as added
  in some extra rules not in recommended group.

Updated testing-library to current versions

arcepaul commented on src/components/atomic/audio-player/audio-player.jsx:
```diff
@@ -59,7 +60,9 @@ const AudioPlayer = ({ audioSrc, lang = 'en', tracking = () => {} }) => {
 				}${hasError ? 'error' : ''}`}
 				onClick={handlePlay}>
-				<span className="show-for-sr">{srText}</span>
+				<span data-testid="span.show-for-sr" className="show-for-sr">
```
weird name for a testid

bryanpizzillo replied:
I was able to remove test ids from the audio-player and the video-player. Some of the other ones are too embedded.

arcepaul commented on src/components/atomic/spinner/__tests__/Spinner.tests.js:
```diff
@@ -1,14 +1,14 @@
-		expect(container.querySelector('.spinkit')).toBeInTheDocument();
-		expect(container.querySelector('.dot1')).toBeInTheDocument();
-		expect(container.querySelector('.dot2')).toBeInTheDocument();
+	it('should have expected button label and fire onRemove handler', () => {
```
A new issue was entered to revisit the spinner and its tests. #35 

fine for this PR

bryanpizzillo committed:
(#33) Fixed Linting Issues

In some cases test ids had to be added to components in order to allow
getBy* rules to pass. This is not optimal, but it is the quickest for
now.

Co-authored-by: Alex <alerosad@users.noreply.github.com>

arcepaul commented on src/components/atomic/InputLabel/__tests__/InputLabel.tests.js:
```diff
@@ -18,11 +18,10 @@ describe('', function () {
-	test('InputLabel renders with correct class names and label hint', function () {
-		const { getByTestId, getByText } = wrapper;
-		const label = getByTestId(`tid-${htmlFor}-label`);
+	it('InputLabel renders with correct class names and label hint', function () {
```
Input Label is redundant here.  I always read the test description as a continuation of the describe text, is that what we should be doing?

seyilonge-nci replied:
I wrote this one. Agreed, it's redundant here. Would be nice we follow "it('should...')" type language.
